superset/connectors/sqla/models.py

```diff
@@ -1438,15 +1440,24 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
                 select_exprs.append(outer)
         elif columns:
             for selected in columns:
+                if isinstance(selected, dict):
```

Comment:
this is the real change, everything else is just setting proper types for columns.

Reply:
also couldnt' use `utils.is_adhoc_column` as mypy can't figure out types properly.

Reply:
there is a typing guard at `utils.core.is_adhoc_column` for this case.